libs/knowledge-store/ragstack_knowledge_store/document_graph.py

```diff
@@ -0,0 +1,114 @@
+from __future__ import annotations
+
+import itertools
+from abc import ABC, abstractmethod
+from typing import Any, Iterable, List, Optional, Union, Iterator, Tuple
+
+from langchain_core.callbacks import CallbackManagerForRetrieverRun
+from langchain_core.documents import Document
+from langchain_core.retrievers import BaseRetriever
+from pydantic import Field
+
+
+def _texts_to_documents(
+    texts: Iterator[str],
+    metadatas: Iterator[dict],
+) -> Iterator[Document]:
```

Comment:
We may want ot consider how we can support diferent kindso f nodes -- especially with things like images and such. We may want a `TextNode` or something like that.

Also, rather than going texts->Document should we introduce the `DocumentNode` or somethnig you had mentioned?

Reply:
I removed DocumentNode because it was close to Document an I felt it didn't add much value. I can add it back.

Reply:
I wonder if we want to have a separate `Node` and then have `DocumentNode` basically be a `Document + Node`. Specifically, this would allow us to introduce `ImageNode`, etc. in the future?

Reply:
You mean we have `add_nodes(nodes: Nodes)` and implementations do type checking to handle the different Node types ? Raising exceptions when they receive a Node type they don't support ?

Reply:
done.PTAL.